bin/cmsl1t

```diff
@@ -0,0 +1,128 @@
+
+TODAY = datetime.now().timetuple()
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+ROOT.gROOT.SetBatch(1)
+ROOT.TH1.SetDefaultSumw2(True)
+separator = '=' * 80
+section = [separator, '{0}', separator]
+section = '\n'.join(section)
+
+
+@timerfunc_log_to(logger.info)
+def process_tuples(config, nevents, analyzers):
+    # Open the data files
+    click.echo(section.format("Loading data"))
+
+    input_files = config.get('input', 'files')
+    click.echo("Input files:")
+
+    if len(input_files) > 10:
+        click.echo(input_files[:10], "... and",
+                   len(input_files) - 10, "more files")
+    else:
+        click.echo(input_files)
+
+    from cmsl1t.playground.eventreader import EventReader
+    reader = EventReader(input_files, events=nevents)
+
+    results = [analyzer.prepare_for_events(reader) for analyzer in analyzers]
+    check(results, analyzers, 'prepare_for_events')
+
+    click.echo(section.format("Processing events"))
+    # Fill the histograms from the tuples
+    counter_rate = nevents / 10 if nevents <= 10000 else 1000
+    for entry, event in enumerate(reader):
+        if entry % counter_rate == 0:
+            click.echo("{} of {}".format(entry, nevents))
+        results = [analyzer.process_event(entry, event)
+                   for analyzer in analyzers]
+        check(results, analyzers, 'process_event')
+        if all(results) is not True:
+            break
+
+
+def run(config, nevents, reload_histograms):
+    results = [False]
+    # Fetch the analyzer
+    analyzers = config.get('analysis', 'analyzers')
+    analyzers = [load_analyzer(analyzer)(config) for analyzer in analyzers]
+    click.echo(analyzers)
+
+    if not reload_histograms:
+        process_tuples(config, nevents, analyzers)
+    else:
+        # TODO: Support multiple input root histogram files
+        pass
+        # isOK = analyzer.reload_histograms(args.inputs[0])
+        # if isOK is not True:
+        #     return "Problem in reload_histograms() with analyzer: " \
+        #         + str(args.analysis)
+    # Write out the histograms
+    for analyzer in analyzers:
+        analyzer.write_histograms()
+
+    # Turn the histograms to plots
+    click.echo(section.format("Making plots"))
+    results = [analyzer.make_plots() for analyzer in analyzers]
+    check(results, analyzers, 'make_plots')
+
+    # Finalize
+    print(section.format("Finalizing things"))
+    results = [analyzer.finalize() for analyzer in analyzers]
+    check(results, analyzers, 'finalize')
+
+    return all(results)
+
+
+def check(results, analyzers, method):
+    msg = 'Problem during {method}() with analyzer "{analyzer}"'
+    if all(results) is not True:
+        for i, r in enumerate(results):
+            if r is not True:
+                click.echo(msg.format(method=method, analyzer=analyzers[i]))
+
+
+@click.command()
+@click.option('-c', '--config_file', help='YAML style config file', type=click.File())
```

Comment:
`config_file` is not really optional right?  I've looked through the click documentation a bit and it doesn't seem like there's an easy way to force an option to be provided and print an error out if it's not.  I'd say we should either check that it's been provided in the code within `analyze()` and print the help / usage out if not, or perhaps better, make it a positional argument?